pkg/utils/utils.go

```diff
@@ -0,0 +1,226 @@
+func GetsriovNumfs(ifName string) (int, error) {
+	var vfTotal int
+
+	sriovFile := filepath.Join(netDirectory, ifName, "device", sriovConfigured)
+	if _, err := os.Lstat(sriovFile); err != nil {
+		return vfTotal, fmt.Errorf("failed to open the sriov_numfs of device %q: %v", ifName, err)
+	}
+
+	data, err := ioutil.ReadFile(sriovFile)
+	if err != nil {
+		return vfTotal, fmt.Errorf("failed to read the sriov_numfs of device %q: %v", ifName, err)
+	}
+
+	if len(data) == 0 {
+		return vfTotal, fmt.Errorf("no data in the file %q", sriovFile)
+	}
+
+	sriovNumfs := strings.TrimSpace(string(data))
+	vfTotal, err = strconv.Atoi(sriovNumfs)
+	if err != nil {
+		return vfTotal, fmt.Errorf("failed to convert sriov_numfs(byte value) to int of device %q: %v", ifName, err)
+	}
+
+	return vfTotal, nil
+}
+
+// GetVfid takes in VF's PCI address(addr) and pfName as string and returns VF's ID as int
+func GetVfid(addr string, pfName string) (int, error) {
+	var id int
+	vfTotal, err := GetsriovNumfs(pfName)
+	if err != nil {
+		return id, err
+	}
+	for vf := 0; vf <= vfTotal; vf++ {
+		vfDir := filepath.Join(netDirectory, pfName, "device", fmt.Sprintf("virtfn%d", vf))
+		_, err := os.Lstat(vfDir)
+		if err != nil {
+			continue
+		}
+		pciinfo, err := os.Readlink(vfDir)
+		if err != nil {
+			continue
+		}
+		pciaddr := pciinfo[len("../"):]
+		if pciaddr == addr {
+			return vf, nil
+		}
+	}
+	return id, fmt.Errorf("unable to get VF ID with PF: %s and VF pci address %v", pfName, addr)
+}
+
+// getSriovPfList returns a list of SRIOV capable PF names as string
+func getSriovPfList() ([]string, error) {
+
+	sriovNetDevices := []string{}
+
+	netDevices, err := ioutil.ReadDir(netDirectory)
+	if err != nil {
+		return sriovNetDevices, fmt.Errorf("Error. Cannot read %s for network device names. Err: %v", netDirectory, err)
+	}
+
+	if len(netDevices) < 1 {
+		return sriovNetDevices, fmt.Errorf("Error. No network device found in %s directory", netDirectory)
+	}
+
+	for _, dev := range netDevices {
+		sriovFilePath := filepath.Join(netDirectory, dev.Name(), "device", "sriov_numvfs")
+
+		if f, err := os.Lstat(sriovFilePath); !os.IsNotExist(err) {
+			if f.Mode().IsRegular() { // and its a regular file
+				sriovNetDevices = append(sriovNetDevices, dev.Name())
+			}
+		}
+	}
+
+	return sriovNetDevices, nil
+}
+
+// GetPfName returns PF net device name of a given VF pci address
+func GetPfName(vf string) (string, error) {
+	pfSymLink := filepath.Join(sysBusPci, vf, "physfn", "net")
+	_, err := os.Lstat(pfSymLink)
+	if err != nil {
+		return "", err
+	}
+
+	files, err := ioutil.ReadDir(pfSymLink)
+	if err != nil {
+		return "", err
+	}
+
+	if len(files) < 1 {
+		return "", fmt.Errorf("PF network device not found")
+	}
+
+	return strings.TrimSpace(files[0].Name()), nil
+}
+
+// GetPciAddress takes in a interface(ifName) and VF id and returns its network interface name as string
+func GetPciAddress(ifName string, vf int) (string, error) {
```

Comment:
Sure, we can do this in next round of refactoring. I wanted to keep the changes minimum and less invasive so that we understand them from existing code(as you can see most changes here are code relocation). Once we have the functions moved in the correct files/pkg then we do further refactoring.